website/static/js/waterbutler.js

```diff
@@ -22,67 +21,58 @@ function getDefaultOptions(path, provider) {
     return options;
 }
 
-function buildUrl(suffix, path, provider, nid, options) {
-    path = path || '/';
-    var baseUrl = window.contextVars.waterbutlerURL + suffix;
-    return baseUrl + $.param($.extend(getDefaultOptions(path, provider), {nid: nid}, options));
-}
-
-var buildCrudUrl = buildUrl.bind(this, 'file?');
-var buildCopyUrl = buildUrl.bind(this, 'copy?');
-var buildMoveUrl = buildUrl.bind(this, 'move?');
-var buildMetadataUrl = buildUrl.bind(this, 'data?');
-var buildRevisionsUrl = buildUrl.bind(this, 'revisions?');
-var buildCreateFolderUrl = buildUrl.bind(this, 'folders?');
-
-
-function buildUploadUrl(path, provider, nid, file, options) {
-    path = (path || '/') + file.name;
-    return buildUrl('file?', path, provider, nid, options);
-}
-
-function buildFromTreebeard(suffix, item, options) {
-    return buildUrl(suffix, item.data.path, item.data.provider, item.data.nodeId, options);
+function getViewOnly() {
+    return $osf.urlParams().view_only;
 }
 
-function buildFromTreebeardFile(item, file, options) {
-    return buildUploadUrl(item.data.path, item.data.provider, item.data.nodeId, file, options);
+function buildFromTreebeard(item, options) {
+    var path;
+    if ('attributes' in item.data) {
+        path = item.data.attributes.path;
+    } else {
+        path = item.data.path;
+    }
+    return buildUrl(path, item.data.provider, item.data.nodeId, options);
 }
 
-function toJsonBlob(item, options) {
-    return $.extend(getDefaultOptions(item.data.path || '/', item.data.provider), {nid: item.data.nodeId}, options);
+function addToken(options) {
+    options = $.extend(options || {}, {token: window.contextVars.accessToken});
+    return options;
 }
 
 module.exports = {
-    toJsonBlob: toJsonBlob,
-    buildDeleteUrl: buildCrudUrl,
-    buildUploadUrl: buildUploadUrl,
-    buildDownloadUrl: function(path, provider, nid, options) {
-        if (window.contextVars.accessToken) {
-            options = $.extend(options || {}, {token: window.contextVars.accessToken});
-        }
-        return buildCrudUrl(path, provider, nid, options);
+    buildDeleteUrl: buildUrl,
+    buildDownloadUrl: buildUrl,
+    buildUploadUrl: function(path, provider, nid, options) {
+        options = $.extend(options || {}, {kind: 'files'});
```

Comment:
`$.extend(options, {foo: 'bar'})` will modify `options` in place.  This means whoever calls the function will have the object they pass in modified.  It would probably be better to do something like:
`var wb_options = $.extend({}, options || {}, {foo: 'bar'});`